2DPlatformerController/Assets/Characters/EnemyCreep.cs

```diff
@@ -92,8 +117,9 @@ private void Attack(IDamagable trgt, Rigidbody2D primaryCollider)
             dmgManager.DistributeDamageWithInvincible(trgt.GetVitalityAttributes(), damagableAttributes, this.damagableAttributes.AttackDamage);
         if (trgt.GetVitalityAttributes().HP <= 0)
```

Comment:
Send inn the vitality attributes and the IDamagablke to the vitality manager and have it destroy the game object instead of doing the HP checking and game object destruction in the character class. So the vitality manager will look at the HP and destroy if necessary